### PR TITLE
Auto-generate distro yamls + docs 

### PR DESCRIPTION
**Pull request by ashwinb**
# What does this PR do?

Automatically generates
- build.yaml
- run.yaml
- run-with-safety.yaml
- parts of markdown docs

for the distributions.

## Test Plan

At this point, this only updates the YAMLs and the docs. Some testing (especially with ollama and vllm) has been performed but needs to be much more tested.

### CODE REVIEW

File: llama_stack/distribution/start_conda_env.sh [hunk 1]
```diff
@@ -33,10 +33,33 @@ shift
 port="$1"
 shift
 
+# Process environment variables from --env arguments
+env_vars=""
```
[the remark this answers is not in the document]
Comment (ashwinb) [reply llama_stack/distribution/start_conda_env.sh:37]: Both ways are actually supported. This just looks exactly the same as the `docker run` command line hence I prefer it. In general, environment variables are detected by modern tools in the following three ways:

- you have a `.env` file which is loaded by something like `dotenv`
- you have them in the environment
- you accept them using `--env` on the CLI

The advantage of the CLI is that you can be very explicit on what you want to be provided to a _certain_ command without polluting anything else.

File: docs/source/getting_started/distributions/self_hosted_distro/together.md [hunk 2]
```diff
@@ -1,62 +1,67 @@
-# Together Distribution
-
-### Connect to a Llama Stack Together Endpoint
-- You may connect to a hosted endpoint `https://llama-stack.together.ai`, serving a Llama Stack distribution
+# Fireworks Distribution
```
Comment (yanxi0830) [inline docs/source/getting_started/distributions/self_hosted_distro/together.md:1]: Wondering which sections gets auto-generated, and should we update the commands to build documentation in [CONTRIBUTING.md](https://github.com/meta-llama/llama-stack/blob/main/CONTRIBUTING.md#building-the-documentation)?

File: llama_stack/providers/remote/inference/ollama/docker_compose.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+services:
```
Comment (yanxi0830) [inline llama_stack/providers/remote/inference/ollama/docker_compose.yaml:1]: Should this `docker_compose.yaml` be inside /providers?

File: llama_stack/providers/remote/inference/tgi/config.py [hunk 4]
```diff
@@ -12,19 +12,20 @@
...
 class TGIImplConfig(BaseModel):
-    host: str = "localhost"
-    port: int = 8080
-    protocol: str = "http"
-
-    @property
-    def url(self) -> str:
-        return f"{self.protocol}://{self.host}:{self.port}"
-
+    url: str = Field(
```
Comment (yanxi0830) [inline llama_stack/providers/remote/inference/tgi/config.py:15]: Do we need to revert back all `<protocol>://<host>:<port>` to `url` for `RemoteProviderConfig`?

**Review by yanxi0830 — APPROVED** [review]
🤞

File: docs/source/getting_started/distributions/self_hosted_distro/together.md (hunk 2 above)
Comment (ashwinb) [reply docs/source/getting_started/distributions/self_hosted_distro/together.md:1]: I added a section to CONTRIBUTING.md but we also need a codegen pre-commit check to ensure there isn't a diff

File: llama_stack/providers/remote/inference/ollama/docker_compose.yaml (hunk 3 above)
Comment (ashwinb) [reply llama_stack/providers/remote/inference/ollama/docker_compose.yaml]: yeah killed it

File: llama_stack/providers/remote/inference/tgi/config.py (hunk 4 above)
Comment (ashwinb) [reply llama_stack/providers/remote/inference/tgi/config.py:15]: yeah we should do that I agree. I will audit that next.

**Pull request merged**